components/match2/wikis/pokemon/match_summary.lua

```diff
@@ -0,0 +1,316 @@
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not epoch=0, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	for gameIndex, game in ipairs(match.games) do
+		local rowDisplay = CustomMatchSummary._createGame(game, gameIndex, match.date)
+		if rowDisplay then
+			body:addRow(rowDisplay)
+		end
+	end
+
+	-- Add Match MVP(s)
+	local mvpInput = match.extradata.mvp
+	if mvpInput then
+		local mvpData = mw.text.split(mvpInput or '', ',')
+		if String.isNotEmpty(mvpData[1]) then
+			local mvp = MatchSummary.Mvp()
+			for _, player in ipairs(mvpData) do
+				if String.isNotEmpty(player) then
+					mvp:addPlayer(player)
+				end
+			end
+
+			body:addRow(mvp)
+		end
+
+	end
+
+	-- Pre-Process Champion Ban Data
+	local championBanData = {}
+	for gameIndex, game in ipairs(match.games) do
+		local extradata = game.extradata
+		local banData = {{}, {}}
+		local numberOfBans = 0
+		for index = 1, _MAX_NUM_BANS do
+			if String.isNotEmpty(extradata['team1ban' .. index]) then
+				numberOfBans = index
+				banData[1][index] = extradata['team1ban' .. index]
+			end
+			if String.isNotEmpty(extradata['team2ban' .. index]) then
+				numberOfBans = index
+				banData[2][index] = extradata['team2ban' .. index]
+			end
+		end
+
+		if numberOfBans > 0 then
+			banData[1].color = extradata.team1side
+			banData[2].color = extradata.team2side
+			banData.numberOfBans = numberOfBans
+			championBanData[gameIndex] = banData
+		end
+	end
+
+	-- Add the Champion Bans
+	if not Table.isEmpty(championBanData) then
+		local championBan = ChampionBan()
+
+		for gameIndex, banData in ipairs(championBanData) do
+			championBan:banRow(banData, gameIndex, banData.numberOfBans, match.date)
+		end
+
+		body:addRow(championBan)
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createGame(game, gameIndex, date)
+	local row = MatchSummary.Row()
+	local extradata = game.extradata or {}
+
+	local championsData = {{}, {}}
+	local championsDataIsEmpty = true
+	for champIndex = 1, _NUM_CHAMPIONS_PICK do
+		if String.isNotEmpty(extradata['team1champion' .. champIndex]) then
+			championsData[1][champIndex] = extradata['team1champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		if String.isNotEmpty(extradata['team2champion' .. champIndex]) then
+			championsData[2][champIndex] = extradata['team2champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		championsData[1].color = extradata.team1side
+		championsData[2].color = extradata.team2side
+	end
+
+	if
+		Table.isEmpty(game.scores) and
+		String.isEmpty(game.winner) and
+		championsDataIsEmpty
+	then
+		return nil
+	end
+
+	row	:addClass('brkts-popup-body-game')
+		:css('font-size', '85%')
+		:css('overflow', 'hidden')
+
+	local score
+	if not Table.isEmpty(game.scores) then
+		score = table.concat(game.scores, '-')
+	end
```

Comment:
Is there more than 2 opponents ever in Pokemon?

Reply:
as far as I've covered events so far, no just 2 individual opponents / teams per matchup